### DISCUSSION TITLE
자바 네이밍 컨벤션에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by gyungchan-jo** [2024-woowacourse-study/level-interview#107] [Clean Code]
.

**Comment by J-I-H-O** [comment]
**패키지** 이름은 소문자로 작성되어야 합니다.
**클래스와 인터페이스 이름**은 대문자로 시작하고, 카멜 케이스로 작성됩니다. 명사 또는 명사구를 사용합니다.
**메서드 이름**은 소문자로 시작하고, 카멜 케이스로 작성됩니다. 동사나 동사구를 사용합니다.
**변수 이름**은 소문자로 시작하고 카멜 케이스로 작성됩니다. 일반적으로 명사나 명사구를 사용합니다.
**상수 이름**은 모두 대문자로 작성하고, 단어 사이에는 밑줄을 사용하여 구분합니다.

**Comment by 3Juhwan** [comment]
- 클래스와 인터페이스: 클래스와 인터페이스의 이름은 대문자로 시작하는 파스칼 표기법을 따릅니다. 예: MyClass, SomeInterface

- 메서드: 메서드의 이름은 소문자로 시작하고 카멜 표기법을 따릅니다. 동사로 시작하여 해당 메서드가 수행하는 동작을 명확히 표현해야 합니다. 예: calculateTotal, getUserInfo

- 변수: 변수의 이름은 소문자로 시작하고 카멜 표기법을 따릅니다. 변수의 역할을 잘 설명할 수 있는 이름을 선택해야 합니다. 예: numberOfStudents, totalAmount

- 상수: 상수의 이름은 모두 대문자로 작성하며 필요에 따라 밑줄(_)로 단어를 구분합니다. 예: MAX_SIZE, PI

- 패키지: 패키지의 이름은 모두 소문자로 작성하며, 필요에 따라 밑줄(_)이나 마침표(.)로 단어를 구분할 수 있습니다. 예: com.example.package1

**Comment by le2sky** [comment]
@3Juhwan  @J-I-H-O 
파스칼 케이스와 카멜 케이스의 차이점은 무엇일까용?

**Reply by J-I-H-O** [reply]
파스칼 케이스는 첫 글자가 대문자로 시작하는 반면, 카멜 케이스는 첫 글자가 소문자로 시작합니다.

자바에서는 클래스 및 인터페이스 네이밍에 파스칼 케이스를 사용하고, 변수 및 메서드명에 카멜 케이스를 사용합니다.

**Reply by le2sky** [reply]
소소하지만,  카멜에는 lower camel과 upper camel로 나뉜다고 합니다.
팀에 따라 upper camel를 pascal이라고 칭하는 조직도 있나본데... 개취 영역인가봐요 

아래 문서에서는 pascal이라는 용어를 사용하네용
https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

카멜만을 이야기 하려면 upper와 lower라는 말을 추가하거나, pascal을 이야기하는 것이 더욱 정확해보입니다.
반박시 땡이말이 맞음

**Comment by GIVEN53** [comment]
클래스 - 파스칼
메서드, 변수 - 카멜
상수 - 대문자, 스네이크
패키지 - 소문자
